rules/no-static-only-class.js

```diff
@@ -0,0 +1,238 @@
+'use strict';
+const {isSemicolonToken} = require('eslint-utils');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const getClassHeadLocation = require('./utils/get-class-head-location');
+const removeSpacesAfter = require('./utils/remove-spaces-after');
+
+const MESSAGE_ID = 'no-static-only-class';
+const messages = {
+	[MESSAGE_ID]: 'Use object instead of class with only static members.'
+};
+
+const selector = [
+	':matches(ClassDeclaration, ClassExpression)',
+	':not([superClass], [decorators.length>0])',
+	'[body.type="ClassBody"]',
+	'[body.body.length>0]'
+].join('');
+
+const assertToken = (token, expected) => {
+	const {type, value} = token;
+	/* istanbul ignore next */
+	if (
+		type !== expected.type ||
+		value !== expected.value
+	) {
+		const issueLink = 'https://github.com/sindresorhus/eslint-plugin-unicorn/issues/new?title=%60no-static-only-class%60%3A%20Unexpected%20token%20error';
+		throw new Error(
+			`Expect token '${JSON.stringify(expected)}', got '${JSON.stringify({type, value})}', Please open an issue at ${issueLink}.`
+		);
+	}
```

Comment:
Can you make this an utility function?

Could be used in these places too:

```
rules/prefer-optional-catch-binding.js
36-					const tokenAfter = context.getTokenAfter(node);
37-
38-					/* istanbul ignore next */
39-					if (!isOpeningParenToken(tokenBefore) || !isClosingParenToken(tokenAfter)) {
40:						throw new Error('Unexpected token.');
41-					}
42-
43-					yield fixer.remove(tokenBefore);
44-					yield fixer.remove(node);

rules/no-array-for-each.js
92-		const returnToken = sourceCode.getFirstToken(returnStatement);
93-
94-		/* istanbul ignore next: `ReturnStatement` firstToken should be `return` */
95-		if (returnToken.value !== 'return') {
96:			throw new Error(`Unexpected token ${returnToken.value}.`);
97-		}
98-
99-		if (!returnStatement.argument) {
100-			yield fixer.replaceText(returnToken, 'continue');
```

Reply:
Done.